pkg/env/certificateRefresher.go

```diff
@@ -0,0 +1,114 @@
+package env
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"crypto/rsa"
+	"crypto/x509"
+	"sync/atomic"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type CertGetter func(context.Context, string) (*rsa.PrivateKey, []*x509.Certificate, error)
+
+type RefreshingCertificateManager interface {
+	StartPoller(context.Context) error
+	StopPoller()
+	GetCertificates() (*rsa.PrivateKey, []*x509.Certificate)
+}
+
+type refreshingCertificate struct {
+	certs    atomic.Value
+	key      atomic.Value
```

Comment:
It seems to me that [`sync.RWMutex`](https://golang.org/pkg/sync/#RWMutex) is more appropriate here, beucase we might see a case of someone reading between these two calls in `pullCertificate`:

```go
r.key.Store(*key)
r.certs.Store(tmpCerts)
```

Another benefit of using a mutext: it eleminates the need to convert between `[]*x509.Certificate` and `[]x509.Certificate`.

Reply:
Thanks for the pointer, you are of course right, I have switched to use RWMutex. Now the race will not happen.